packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -49,41 +49,53 @@ const fixId = id => {
 }
 exports.fixId = fixId
 
+const shouldBeWalked = (object, alreadyWalkedObjectRefs) =>
+  object && typeof object === `object` && !alreadyWalkedObjectRefs.has(object)
+
 const fixIds = object => _fixIds(object)
```

Comment:
This "entry" function made sense when it was recursive but now it's just irrelevant overhead. Would you mind removing it? Just drop this line and rename `_fixIds` to `fixIds`.